src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -372,6 +376,8 @@ private AIState lookForNeededItems()
 
     /**
      * Updates the itemsCurrentlyNeeded with current values.
+     * 
+     * This function seems bugged, as it gives itemsCurrentlyNeeded = isEmpty 
```

Comment:
`syncNeededItemsWithInventory()` should refill `itemsCurrentlyNeeded` 
Before the function, iCN contains a stack of items but afterwards its empty.
```
job.clearItemsNeeded();
        itemsNeeded.forEach(job::addItemNeeded);
        InventoryUtils.getInventoryAsList(worker.getInventoryCitizen()).forEach(job::removeItemNeeded);
        itemsCurrentlyNeeded = new ArrayList<>(job.getItemsNeeded()); //<- This seems not to work
```

unfortunately I am a totally beginner so I was not able to resolve this. this is why I wrote the "workaround"